ff/src/bits.rs

```diff
@@ -0,0 +1,82 @@
+/// Iterates over a slice of `u64` in *big-endian* order.
+#[derive(Debug)]
+pub struct BitIteratorBE<Slice: AsRef<[u64]>> {
+    s: Slice,
+    n: usize,
+}
+
+impl<Slice: AsRef<[u64]>> BitIteratorBE<Slice> {
+    pub fn new(s: Slice) -> Self {
+        let n = s.as_ref().len() * 64;
+        BitIteratorBE { s, n }
+    }
+
+    /// Construct an iterator that automatically skips any leading zeros.
+    /// That is, it skips all zeros before the most-significant one.
+    pub fn without_leading_zeros(s: Slice) -> impl Iterator<Item = bool> {
+        Self::new(s).skip_while(|b| !b)
+    }
+}
+
+impl<Slice: AsRef<[u64]>> Iterator for BitIteratorBE<Slice> {
+    type Item = bool;
+
+    fn next(&mut self) -> Option<bool> {
+        if self.n == 0 {
+            None
+        } else {
+            self.n -= 1;
+            let part = self.n / 64;
+            let bit = self.n - (64 * part);
+
+            Some(self.s.as_ref()[part] & (1 << bit) > 0)
+        }
+    }
+}
+
+/// Iterates over a slice of `u64` in *little-endian* order.
+#[derive(Debug)]
+pub struct BitIteratorLE<Slice: AsRef<[u64]>> {
+    s: Slice,
+    n: usize,
+    max_len: usize,
+}
+
+impl<Slice: AsRef<[u64]>> BitIteratorLE<Slice> {
+    pub fn new(s: Slice) -> Self {
+        let n = 0;
+        let max_len = s.as_ref().len() * 64;
+        BitIteratorLE { s, n, max_len }
+    }
+
+    /// Construct an iterator that automatically skips any trailing zeros.
+    /// That is, it skips all zeros after the most-significant one.
```

Comment:
```suggestion
    /// That is, it skips all zeros after the most significant "one."
```

Same as above. But this is pending for discussion.